src/Compilers/Core/CodeAnalysisTest/Diagnostics/CompilationWithAnalyzersTests.cs

```diff
@@ -7,17 +7,170 @@
 using System.Threading;
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.VisualBasic;
 using Roslyn.Test.Utilities;
 using Roslyn.Utilities;
 using Xunit;
 using static Microsoft.CodeAnalysis.CommonDiagnosticAnalyzers;
+using static Roslyn.Test.Utilities.TestHelpers;
 
 namespace Microsoft.CodeAnalysis.UnitTests.Diagnostics
 {
     using SimpleDiagnostic = Diagnostic.SimpleDiagnostic;
 
     public class CompilationWithAnalyzersTests : TestBase
     {
+        private static CSharpCompilation CreateCSharpCompilation(
+            string source,
+            IEnumerable<MetadataReference> references = null,
+            CSharpCompilationOptions options = null)
+            => CSharpCompilation.Create(
+                Guid.NewGuid().ToString(),
+                SpecializedCollections.SingletonEnumerable(CSharp.SyntaxFactory.ParseSyntaxTree(source)),
+                references ?? SpecializedCollections.SingletonEnumerable(MscorlibRef),
+                options);
+
+        private static VisualBasicCompilation CreateBasicCompilation(
+            string source,
+            IEnumerable<MetadataReference> references = null,
+            VisualBasicCompilationOptions options = null)
+            => VisualBasicCompilation.Create(
+                Guid.NewGuid().ToString(),
+                SpecializedCollections.SingletonEnumerable(VisualBasic.SyntaxFactory.ParseSyntaxTree(source)),
+                references ?? SpecializedCollections.SingletonEnumerable(MscorlibRef),
+                options);
+
+        [Fact]
+        public void CSharpGetEffectiveDiagnosticsPragmaSuppressed()
+        {
+            var options = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary);
+            var c = CreateCSharpCompilation(@"
+class C
+{
+#pragma warning disable CS0169
+    int _f;
+#pragma warning restore CS0169
+}", options: options);
+            CompilationWithAnalyzers.GetEffectiveDiagnostics(c.GetDiagnostics(), c).Verify();
+
+            options = options.WithReportSuppressedDiagnostics(true);
+            c = c.WithOptions(options);
+            CompilationWithAnalyzers.GetEffectiveDiagnostics(c.GetDiagnostics(), c).Verify(
```

Comment:
@agocke, actually looking @AlekseyTs comment _Should we adjust this comment? It looks like there might be another way to mark diagnostics as suppressed now._, I now recall that the original behavior was as per the required design of the [ReportSuppressedDiagnostics](http://source.roslyn.io/#Microsoft.CodeAnalysis/Compilation/CompilationOptions.cs,a827de5c55dd2796,references) flag and this change is a **breaking change**. 

This flag was added for the IDE feature that enables showing _source suppressed_ instances of a diagnostic  in the error list when the error list column _Suppression State_ is changed to `Suppressed`. The desired behavior was to _only_ show diagnostics that are _enabled_ for the compilation, not the ones that are _disabled_ either via a `nowarn` or a ruleset entry with action `None`.

Reply:
Following was the reasoning behind the original design of `ReportSuppressedDiagnostics`: There is a subtle difference between _disabling_ a diagnostic for the compilation and _suppressing_ specific diagnostic instances. Auditing the latter set is valuable to identify source suppressions for diagnostics that are considered _valuable_ for the compilation (otherwise the diagnostic would have been disabled), and should possibly be fixed and unsuppressed, if not false positives. Viewing the former set is not really useful, otherwise user would not have disabled the diagnostic ID for the compilation.